src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -482,16 +485,40 @@ def _get_data(
                 # This should only happen for payment status update,
                 # where only a subset of the variables are updated.
                 continue
-            target_path = mapping["target_path"]
 
-            # Type hint is wrong: currently some static variables are of type date/datetime
             value = cast(Any, variables_values[variable_key])
 
             # Comply with JSON Schema "format" specs:
             if isinstance(value, (datetime, date)):
                 value = value.isoformat()
 
-            glom.assign(record_data, glom.Path(*target_path), value, missing=dict)
+            target_path = mapping.get("target_path")
+            if (
+                component_key_type_mappings
+                and component_key_type_mappings.get(variable_key) == "addressNL"
+            ):
```

Comment:
this is not yet a proper generic solution because it requires us to write `if component_type == '...'` code which gets very messy very quickly, so please put a `TODO` comment saying this will need to be addressed at some point